purchases/src/main/kotlin/com/revenuecat/purchases/common/diagnostics/DiagnosticsTracker.kt

```diff
@@ -199,6 +202,36 @@ internal class DiagnosticsTracker(
         trackEvent(event)
     }
 
+    // region Offline Entitlements
+
+    fun trackEnteredOfflineEntitlementsMode() {
+        val event = DiagnosticsEntry(
+            name = DiagnosticsEntryName.ENTERED_OFFLINE_ENTITLEMENTS_MODE,
+            properties = mapOf(),
+        )
+        trackEvent(event)
+    }
+
+    fun trackErrorEnteringOfflineEntitlementsMode(error: PurchasesError) {
+        val isOneTimePurchaseFoundError = error.code == PurchasesErrorCode.UnsupportedError &&
+            error.underlyingErrorMessage == OfflineEntitlementsStrings.OFFLINE_ENTITLEMENTS_UNSUPPORTED_INAPP_PURCHASES
+        val reason = if (isOneTimePurchaseFoundError) {
+            "one_time_purchase_found"
+        } else {
+            "unknown"
+        }
+        val event = DiagnosticsEntry(
+            name = DiagnosticsEntryName.ERROR_ENTERING_OFFLINE_ENTITLEMENTS_MODE,
+            properties = mapOf(
+                "offline_entitlement_error_reason" to reason,
+                "error_message" to "${error.message}. Underlying error: ${error.underlyingErrorMessage}",
```

Comment:
This won't actually be tracked in Grafana, but I thought it would still be useful to send it.

Reply:
at the very least it'll be useful in elastic if we're looking through logs of a single customer